pytorch_lightning/trainer/callback_hook.py

```diff
@@ -96,7 +96,15 @@ def on_train_epoch_end(self, outputs: EPOCH_OUTPUT):
             outputs: List of outputs on each ``train`` epoch
         """
         for callback in self.callbacks:
-            callback.on_train_epoch_end(self, self.lightning_module, outputs)
+            if is_param_in_hook_signature(callback.on_train_epoch_end, "outputs"):
```

Comment:
What is users call it differently ?

Reply:
Better to check the signature in different ways to ensure robustness.

Reply:
n00b question: suggestions on what other ways we can check the signature? would *args or **kwargs matter here? I saw other callbacks using *args to indicate they didn't use any additional positional arguments the trainer was sending the callback, and I didn't want to mistakenly send these epoch outputs

Reply:
I think this is fine personally, especially since this is tied to a deprecation path.